apps/hash-ai-worker-ts/src/activities/shared/embeddings.ts

```diff
@@ -0,0 +1,105 @@
+import type { VersionedUrl } from "@blockprotocol/type-system";
+import type { PropertyType } from "@local/hash-graph-client";
+import type {
+  BaseUrl,
+  EntityPropertiesObject,
+  EntityPropertyValue,
+} from "@local/hash-subgraph";
+import { extractBaseUrl } from "@local/hash-subgraph/type-system-patch";
+import OpenAI from "openai";
+
+import Usage = OpenAI.CreateEmbeddingResponse.Usage;
+
+const openai = new OpenAI({
+  apiKey: process.env.OPENAI_API_KEY,
+});
+
+export const createEmbeddings = async (params: { input: string[] }) => {
+  const response = await openai.embeddings.create({
+    input: params.input,
+    model: "text-embedding-ada-002",
+  });
+
+  return {
+    usage: response.usage,
+    embeddings: response.data.map((data) => data.embedding),
+  };
+};
+
+const createPropertyEmbeddingInput = (params: {
+  propertyTypeSchema: Pick<PropertyType, "title">;
+  propertyValue: EntityPropertyValue;
+}): string => {
+  return `${params.propertyTypeSchema.title}: ${typeof params.propertyValue === "string" ? params.propertyValue : JSON.stringify(params.propertyValue)}`;
+};
+
+/**
+ * Creates embeddings for (a) each property and finally (b) a new line-separated list of all properties.
+ */
+export const createEntityEmbeddings = async (params: {
+  entityProperties: EntityPropertiesObject;
+  propertyTypes: { title: string; $id: VersionedUrl }[];
+}): Promise<{
+  embeddings: {
+    /** If 'property' is absent, this is the combined embedding for all properties */
+    property?: BaseUrl;
+    embedding: number[];
+  }[];
+  usage: Usage;
+}> => {
```

Comment:
This is the only real change in this file – I changed the signature a little to make it easier to call from elsewhere (it doesn't need all the metadata etc it was previously typed as needing)